Update MemoryPlanning Verifier to only assume model has user input if it has at least one tensor input

jhelsby opened the pull request:
Fixes #10522

### Summary

MemoryPlanning verifier currently blows up if all the user inputs are prims. #10522 suggested:
> We need to improve the logic of the _do_user_inputs_exist to probably just return false if all the inputs are prims.

This PR implements this suggestion, with accompanying unit tests. `_do_user_inputs_exist` now returns True if it has at least one tensor input, and False otherwise.

### Test plan

Added unit tests to `test_memory_planning.py` and ran them with 
```bash
pytest exir/tests/test_memory_planning.py::test_memory_planning.py
```
Please note you must comment out the line 69 from `/pytest.ini` for this to work:

```
--ignore=exir/tests/test_memory_planning.py
```

On my machine, I also had to comment out line 60 from `test_memory_planning.py` for the test to run without errors.

```
torch.ops.load_library("//executorch/kernels/portable:custom_ops_generated_lib")
```

[The tests I wrote aren't dependent on this library.]

jhelsby commented:
@pytorchbot label "release notes: exir"

digantdesai commented:
Started CI (Need to do for new, external contributions). Rebase please. 

Also marked it as Ready to signal reviewers that it's, well, ready for a review. Thanks for the contribution. Welcome to Executorch 😊

jhelsby commented:
Thank you, I just rebased. I also fixed the linting issues which the CI tests identified.

jhelsby committed:
Update MemoryPlanning Verifier to only assume model has user input if it has at least one tensor input

MemoryPlanning verifier currently blows up if all the user inputs are prims.
This change means its helper function _do_user_inputs_exist returns
false
if all its inputs are prims. It only returns true if at least one
input is a tensor.

digantdesai commented:
cc @JacobSzwejbka  can you take a look? 🙏

larryliu0820 commented:
@JacobSzwejbka gentle ping

JacobSzwejbka reviewed (changes requested):
I dont see any obvious issue with the code, the all red ci though made me think it was still under iteration.

jhelsby committed:
Merge branch 'main' into bugfix/memoryplanningverifier/douserinputsexist-noprims

jhelsby commented:
Many thanks for looking at this @JacobSzwejbka. Is there anything I can do now to help get this PR closed?

larryliu0820 commented:
Kicked off CI jobs.

larryliu0820 commented:
I don't think there's any other comment so approving

jhelsby committed:
Merge branch 'main' into bugfix/memoryplanningverifier/douserinputsexist-noprims

jhelsby commented:
I'm so sorry @larryliu0820, I accidentally merged the branch which seems to have stopped the running workflows. Could you please kick off the CI jobs again?

jhelsby commented:
Fantastic, thank you so much for your help and patience throughout! This was my first code contribution to a major open-source project. 😄
